account-kit/signer/src/base.ts

```diff
@@ -836,6 +864,53 @@ export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
     return new SolanaSigner(this.inner);
   };
 
+  private authenticateWithEmailNew = async (
+    params: Extract<AuthParams, { type: "email" }>
+  ): Promise<AuthStepResult> => {
+    if ("bundle" in params) {
+      const user = await this.completeEmailAuth(params);
+      return {
+        status: AlchemySignerStatus.CONNECTED,
+        user,
+      };
+    }
+
+    if (!("email" in params)) {
```

Comment:
This check isn't strictly necessary because the type system should supposedly enforce that `email` is present in this branch, but doesn't hurt I suppose.